File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        sorted(nums)

        left = 0
        ans = float('inf')
        total = 0
        for right in range(len(nums)):
            total += nums[right]

            while total >= target:
                ans = min(ans, right-left+1)
                total -= nums[left]
                left += 1
        
        return 0 if ans == float('inf') else ans
                
            

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {

        int left = 0;
        int ans = Integer.MAX_VALUE;
        int total = 0;
        for(int right = 0; right < nums.length; right++){
            total = total + nums[right];

            while(total >= target) {

                ans = Math.min(ans, right-left+1);
                total = total - nums[left];
                left++;
            }
        }
        return (ans == Integer.MAX_VALUE) ? 0 : ans;
    }
}